### PR TITLE
[backend] Fix dependency sorting in mustWrite

### PR DESCRIPTION
**Pull request by pgavlin**
Just use slices.Sort instead of sort.Slice.

### CODE REVIEW

**Comment by pgavlin** [comment]
Will add tests for this tomorrow. Without this we'll likely have major performance regressions for no-op updates.

File: pkg/backend/snapshot.go [hunk 1]
```diff
@@ -271,7 +272,7 @@ func (ssm *sameSnapshotMutation) mustWrite(step *deploy.SameStep) bool {
 		sortDeps := func(deps []resource.URN) []resource.URN {
 			result := make([]resource.URN, len(deps))
 			copy(result, deps)
-			sort.Slice(result, func(i, j int) bool { return deps[i] < deps[j] })
+			slices.Sort(result)
```
Comment (justinvp) [inline pkg/backend/snapshot.go:275]: Is this because `sort.Slice` isn't stable and `slices.Sort` is?

Comment (justinvp) [reply pkg/backend/snapshot.go:275]: Oh, I see `result` vs `deps`.

Comment (justinvp) [reply pkg/backend/snapshot.go:275]: I just did an audit of all the other uses of `sort.Slice` in the repo, and there aren't any other bugs like this. 👍

**Review by tgummerer — APPROVED** [review]

**Comment by tgummerer** [comment]
> Probably worth merging this as is and @tgummerer and @julienp can probably get a patch release out now during EU hours. Then add tests in US time after.

Happy to get a patch release out.  I feel like this should have a changelog though if it's worth a patch release.  I'll remove it from the queue add something and then re-add it.

**Pull request merged**